packages/core/src/transform/template/rewrite-module.ts

```diff
@@ -116,15 +159,77 @@ function parseScript(ts: TSLib, script: SourceFile, environment: GlintEnvironmen
     void emitMetadata.set(node, Object.assign(emitMetadata.get(node) ?? {}, data));
 
   let { preprocess, transform } = environment.getConfigForExtension(extension) ?? {};
-  let preprocessed = preprocess?.(contents, filename) ?? { contents };
+
+  let original: {
+    contents: string;
+    data?: {
+      // SAFETY: type exists elsewhere (the environments)
+      templateLocations: any[];
+    };
+  } = { contents, data: { templateLocations: [] } };
+
+  let preprocessed = original;
+  let error: ParseError | undefined;
+
+  try {
+    preprocessed = preprocess?.(contents, filename) ?? original;
+  } catch (e) {
+    if (typeof e === 'object' && e !== null) {
+      // Parse Errors from the rust parser
+      if ('source_code' in e) {
```

Comment:
It may be useful for content-tag to emit this error as a JSON object or something so I don't have to parse the error-string here.